lib/document.js

```diff
@@ -167,6 +165,10 @@ function Document(obj, fields, skipId, options) {
     $__applyDefaults(this, fields, skipId, exclude, hasIncludedChildren, false, options.skipDefaults, {
       isNew: this.isNew
     });
+  } else if (!defaults) {
+    EventEmitter.prototype.once.call(this, 'init', () => {
```

Comment:
I'm not sure why you need this - if `defaults` correctly propagates down from the query's `_mongooseOptions`, you shouldn't need this. Why are defaults still getting set if `defaults` is false?

Reply:
I tried negating the defaults if statement to see if that would solve it but to no avail. The defaults are already set because when mongoose creates the document, if there are values with defaults it inserts them with the default values. So the only way to get rid of them it seems is to remove them after the fact.